test/run_tests.sh

```diff
@@ -208,6 +208,7 @@ function run_xla_op_tests3 {
   run_test "$CDIR/test_input_output_aliases.py"
   run_test "$CDIR/test_torch_distributed_xla_backend.py"
   run_torchrun "$CDIR/pjrt/test_torchrun.py"
+  run_test "$CDIR/test_persistent_cache.py"
```

Comment:
Thanks @jonb377! I think we want to test 3 different cases
1. single device
2. multi-device non spmd
3. multi-device spmd
and in later time we want to at least manually test in pod.

Reply:
Sure! Right now, this covers 1 and 2. I'll add an MP test to the suite.

Reply:
Just a heads up, the tests are pretty slow since we have to boot up the runtime multiple times (right now it's ~80seconds, with MP it will be 2min).

Reply:
Done - the tests now include:
- Single-device
- Replicated SPMD
- Sharded SPMD
- MP